cosmo: feat: “traits” module (#17702)
astropy/astropy: docs/glossary.rst

      :class:`~astropy.time.Time`, e.g. `str`, array-like[str], `~datetime.datetime`, or
      `~numpy.datetime64`.

   trait type
      In short, a trait type is an orthogonal mixin class. Strictly following the

>       definition in :mod:`mypyc`, a trait type is not a Python type, but a C type.

Reviewer: Sphinx can't figure out the links to Mypyc. Maybe it's best not to mention Mypyc here at all. The glossary entry should be more focused on traits that are not decorated with `@trait`. The glossary entry not really needed for the decorated traits because it would be simple enough to refer to the decorator documentation.

Author: Done.

>       In short, a trait type is an orthogonal mixin class. Strictly following the
>       definition, a trait type is not a Python type, but a C type. However, compiled
>       Python types are often not compiled to C types, so trait types can be Python types
>       as well. A Python trait type has the following properties:

Reviewer: Why do you need to talk about Python types and C types here at all?
```suggestion
      In short, a trait type is a class with the following properties:
```
